test/metalearn/metafeatures/test_metafeatures.py

```diff
@@ -20,78 +22,97 @@ class MetaFeaturesWithDataTestCase(unittest.TestCase):
 
     def setUp(self):
         self.datasets = {}
-        self.data_folder = './test/data/'
+        self.data_folder = "./test/data/"
         with open(self.data_folder + "test_dataset_metadata.json", "r") as fh:
             dataset_descriptions = json.load(fh)
 
-        for dataset_metadata in dataset_descriptions:
-            filename = dataset_metadata["filename"]
-            target_class_name = dataset_metadata["target_class_name"]
-            index_col_name = dataset_metadata.get('index_col_name', None)
+        for dataset_description in dataset_descriptions:
+            filename = dataset_description["filename"]
+            target_class_name = dataset_description["target_class_name"]
+            index_col_name = dataset_description.get("index_col_name", None)
             X, Y, column_types = read_dataset(filename, index_col_name, target_class_name)
             known_dataset_metafeatures_path = get_dataset_metafeatures_path(filename)
             if os.path.exists(known_dataset_metafeatures_path):
                 with open(known_dataset_metafeatures_path) as fh:
                     metafeatures = json.load(fh)
-                self.datasets[filename] = {"X": X, "Y": Y, "metafeatures": metafeatures, "path": known_dataset_metafeatures_path}
+                self.datasets[filename] = {
+                    "X": X, "Y": Y, "known_metafeatures": metafeatures,
+                    "known_metafeatures_path": known_dataset_metafeatures_path
+                }
             else:
-                raise FileNotFoundError(f'{known_dataset_metafeatures_path} does not exist' )
-            
+                raise FileNotFoundError(f"{known_dataset_metafeatures_path} does not exist" )
+
     def tearDown(self):
         del self.datasets
 
-    def process_result(self, results, test_name):
```

Comment:
Would it be useful to keep a function like this? Without it _test_correctness quits as soon as a dataset fails and we potentially lose information about future datasets. Also, I we have additional _tests in the future such as compare_mf_lists or compute_side_effects, could we possibly process them all in this function at the same time?

Reply:
That is a good point about other datasets. It would be helpful if we know whether the error is dataset specific.

As far as additional tests after testing correctness, if the metafeatures are not computed correctly,  there are few tests that can be considered valid. It may be a waste of time to run other tests when we know the code needs to change/be fixed. Once we fix the code after a failed correctness test, we will have to run the tests again and eventually they will be run.